### PR TITLE
Add DhtClient ctors accepting Neo.Config arguments

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
Blocked until swarm v4.3.0 is released.

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
@scott-gibson-sociomantic please peruse.

**Comment by gavin-norman-sociomantic** [comment]
Based on #42. Still blocked.

**Comment by gavin-norman-sociomantic** [comment]
#42 merged. Unblocked.

File: src/dhtproto/client/DhtClient.d [hunk 1]
```diff
@@ -711,30 +783,65 @@ public class DhtClient : IClient
...
+
+
+    /***************************************************************************
+
         Constructor with support for the neo and legacy protocols. This
         constructor that accepts all arguments manually (i.e. not read from
         config files) is mostly of use in tests.
 
         Params:
-            epoll = select dispatcher to use
```
Comment (nemanja-boric-sociomantic) [inline src/dhtproto/client/DhtClient.d]: Rest of the changes in this file should go in the doc tidy-up commit.

File: src/dhtproto/client/mixins/NeoSupport.d [hunk 2]
```diff
@@ -1291,7 +1291,42 @@ template NeoSupport ( )
...
+                re-established) and when the hash-range for a node is first
+                queried. Of type:
+                void delegate ( Neo.DhtConnNotification )
+
+    ***************************************************************************/
+
+    private void neoInit ( Neo.Config config,
+        Neo.DhtConnectionNotifier user_conn_notifier )
+    {
+        this.user_conn_notifier = user_conn_notifier;
```
Comment (nemanja-boric-sociomantic) [inline src/dhtproto/client/mixins/NeoSupport.d]: The old overload of `neoInit` has a contract asserting that the `user_conn_notifier` must be not-null. Is this not applicable here?

File: src/dhtproto/client/DhtClient.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/dhtproto/client/DhtClient.d]: Ah yeah, I see it's got a bit mixed up.

File: src/dhtproto/client/mixins/NeoSupport.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/dhtproto/client/mixins/NeoSupport.d]: Good catch. (I think I originally intended the new one to call the old one, but it didn't work out that way.)

**Comment by gavin-norman-sociomantic** [comment]
Updated:
* Moved doc changes into correct commit.
* Added in contract to `neoInit` overload.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**